stories/__tests__/SQFormRadioButtonGroup.stories.test.js

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import {composeStories} from '@storybook/testing-react';
+import {render, screen, waitFor} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import * as stories from '../SQFormRadioButtonGroup.stories';
+
+const {
+  Default: SQFormRadioButtonGroup,
+  WithValidation: SQFormRadioButtonGroupWithValidation
+} = composeStories(stories);
+
+describe('SQFormRadioButtonGroup Tests', () => {
+  describe('Without validation', () => {
+    it('renders a label and group of radio buttons', () => {
+      render(<SQFormRadioButtonGroup size="auto" />);
+
+      const label = screen.getByText(/Pandas/i);
+      expect(label).toBeInTheDocument();
+      expect(label).toHaveTextContent('Pandas');
+
+      const radioGroup = screen.getByRole('radiogroup', {name: /pandas/i});
+      expect(radioGroup).toBeInTheDocument();
+    });
+
+    describe('radio button group', () => {
+      it('renders three radio buttons', () => {
+        render(<SQFormRadioButtonGroup size="auto" />);
+
+        const radioButtons = screen.getAllByRole('radio');
+        expect(radioButtons).toHaveLength(3);
+      });
+
+      it('has radio buttons with proper labels', () => {
+        render(<SQFormRadioButtonGroup size="auto" />);
+
+        const radioButtons = screen.getAllByRole('radio');
+
+        const giantPandaButton = radioButtons[0];
+        const giantPandaLabel = screen.getByText(/Giant Panda/i);
+
+        const redPandaButton = radioButtons[1];
+        const redPandaLabel = screen.getByText(/Red Panda/i);
+
+        const kungFuPandaButton = radioButtons[2];
+        const kungFuPandaLabel = screen.getByText(/Kung Fu Panda/i);
+
+        expect(giantPandaButton).toBeInTheDocument();
+        expect(giantPandaLabel).toBeInTheDocument();
+        expect(giantPandaLabel).toHaveTextContent('Giant Panda');
+
+        expect(redPandaButton).toBeInTheDocument();
+        expect(redPandaLabel).toBeInTheDocument();
+        expect(redPandaLabel).toHaveTextContent('Red Panda');
+
+        expect(kungFuPandaButton).toBeInTheDocument();
+        expect(kungFuPandaLabel).toBeInTheDocument();
+        expect(kungFuPandaLabel).toHaveTextContent('Kung Fu Panda');
+      });
+    });
+
+    describe('Selection', () => {
+      it('allows selecting button', () => {
+        render(<SQFormRadioButtonGroup size="auto" />);
+        const firstRadioButton = screen.getAllByRole('radio')[0];
+
+        expect(firstRadioButton).not.toBeChecked();
+        userEvent.click(firstRadioButton);
+        expect(firstRadioButton).toBeChecked();
+      });
+
+      it('allows selection via label', () => {
+        render(<SQFormRadioButtonGroup size="auto" />);
+
+        const giantPandaRadioButton = screen.getAllByRole('radio')[0];
```

Comment:
Since you're matching this input with clicked label text, it would be better to get it via `getByLabelText` rather than it's order in the document